stringdtype/stringdtype/src/dtype.c

```diff
@@ -113,16 +114,20 @@ stringdtype_setitem(StringDTypeObject *descr, PyObject *obj, char **dataptr)
         return -1;
     }
 
-    *dataptr = malloc(sizeof(char) * length + 1);
-    strncpy(*dataptr, val, length + 1);
+    ss *str_val = ssnewlen(val, length);
+    if (str_val == NULL) {
+        PyErr_SetString(PyExc_MemoryError, "ssnewlen failed");
+        return -1;
+    }
+    *dataptr = (char *)str_val;
```

Comment:
Is this a memory leak? How do I tell the difference between when I'm filling empty invalid data and when I'm overriding an existing string that I need to free first.

Reply:
Set the `NEEDS_INIT` flag and check for `NULL` (currently, I only guarantee NULLing of data as initialization, I somewhat hope that is sufficient).